server/push_test.go

```diff
@@ -95,15 +89,40 @@ func TestPush_ShouldReturn422ForNonRetryable(t *testing.T) {
 
 	s.rtmpConnections["mani"] = cxn
 
-	req.Header.Set("Accept", "multipart/mixed")
+	// Should return 503 if got a retryable error with no sessions to use
```

Comment:
I was thinking about whether there could be a case where B runs out of sessions before it hits the max # of attempts. Dumping some thoughts below just for visibility.

Based on https://github.com/livepeer/go-livepeer/blob/fbcdccb6ca545ec68f8a6bd533f3d843ec6ad9b1/server/broadcast.go#L775 I believe that if B runs out of sessions, it will still hit the max # of attempts - each of the remaining attempts will just result in an error caused by B being out of sessions. And since we check for a non-retryable error [here](https://github.com/livepeer/go-livepeer/blob/fbcdccb6ca545ec68f8a6bd533f3d843ec6ad9b1/server/mediaserver.go#L959) and return before the [no sessions available error check](https://github.com/livepeer/go-livepeer/blob/fbcdccb6ca545ec68f8a6bd533f3d843ec6ad9b1/server/mediaserver.go#L974) in the HTTP push handler we will always return a 422 in this scenario before getting to the code that triggers a 503.